specs/refactor_gluster_integration_get_state_dump.adoc

```diff
@@ -0,0 +1,235 @@
+= Refactor Gluster Integration's state dump module
+
+
+The gluster get-state output is consumed by tendrl-gluster-integration. We need
+to re-factor the way Tendrl consumes and processes the get-state output.
+
+Notes:
+
+* In light of new changes to gluster get-state
+  http://review.gluster.org/#/c/15889
+
+
+== Problem description
+
+Current Tendrl way of consuming and processing the gluster get-state output
+considers only the volumes and bricks and pushes the details to central store.
+It needs to be modified to update the newly available volume options as well to
+central store.
+
+== Use Cases
+
+This addresses the use case of obtaining refreshed gluster cluster state and
+updating the same to central store.
+
+== Proposed change
+
+* Re-factor the code to handle the current output structure of get-state command
+related to volume options
+
+The current output from gluster get-state command is proposed to be of the
+format
+
+```
+[Global]
+MYUUID: 3780fd01-8047-402c-81df-e361a1935bb8
+op-version: 30900
+
+[Global options]
+
+[Peers]
+Peer1.primary_hostname: <fqdn1>
+Peer1.uuid: 3c94c637-cb3b-4580-b718-a7fc5abef4ef
+Peer1.state: Peer in Cluster
+Peer1.connected: Connected
+Peer1.othernames:
+Peer2.primary_hostname: <fqdn2>
+Peer2.uuid: 3f1459e9-9f68-4c6c-8a30-53c09188cd5f
+Peer2.state: Peer in Cluster
+Peer2.connected: Connected
+Peer2.othernames:
+
+[Volumes]
+Volume1.name: vol1
+Volume1.id: 5e24f6ed-239e-4961-ac47-1e9ed44aa7b1
+Volume1.type: Distribute
+Volume1.transport_type: tcp
+Volume1.status: Started
+Volume1.brickcount: 3
```

Comment:
As long as I understand we should define one attribute "options" for the volume object in definitions which should be a dictionary. If dictionary is not supported we need to look at that and see how that could be supported.

This attribute "options" could be involved in below scenarios -
1. Create Volume: Volume volume create only default options would be set by gluster and post creation when volume gets refreshed in tendrl, it would get the dict with default keys in "options" dictionary.
2. Edit Volume: During edit volume there are two options. Either change the value of the existing option or add new one. Adding new one is more driven by enabling a new feature for volume rather than manually adding the key and value in "options"
3. Listing of volume: The api needs to get the "options" as well, the way they get the bricks and append to the output as a dict

I feel, as long as definitions support dictionary type it should be achievable and more generic in nature. Otherwise in future when gluster adds another feature, we need to update tendrl code again and again only for this reason.

@nthomas-redhat @brainfunked comments??